tests/unit/test_projection_engine_integration.py

```
@@ -0,0 +1,94 @@
+from app.infrastructure.services.projection_engine import ProjectionEngine
+
+
+@pytest.mark.asyncio
+async def test_projection_engine_uses_skip_locked_and_subject_type_filtering(async_session):
+    """Smoke test: engine can run one cycle and call get_events_since_seq with subject_type."""
+    repo = ProjectionRepository(async_session)
+    event_repo = EventRepository(async_session)
+    registry = ProjectionRegistry()
+
+    # Register a dummy projection handler.
+    async def handler(state: dict, event):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `handler`

(ANN202)